### ISSUE TITLE
Repeatedly triggering material 3 ink sparkle leads to permanent raster slowdown

### ISSUE THREAD

**Issue by Elblassy** [flutter/flutter#104545]
## Details

When i use material 3 is true and press to icon button many times in list all view get lagged and drop frame happened to 28hz

https://user-images.githubusercontent.com/30121703/170135403-3c6b312c-77c0-44ef-a8df-69f242fe1043.mp4

**ِAndroid:**
**Android 12:**
**Tested on mi9t and samsung A22:**

```
```

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.0.1, on macOS 11.5.2 20G95 darwin-x64, locale
    en-EG)
[✓] Android toolchain - develop for Android devices (Android SDK version
    32.1.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[!] Android Studio
    ✗ Unable to find bundled Java version.
[✓] IntelliJ IDEA Community Edition (version 2021.3)
[✓] VS Code (version 1.67.0)
[✓] VS Code (version 1.53.2)
[✓] Connected device (5 available)
[✓] HTTP Host Availability

```
```

**Tested Code**
```
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        useMaterial3: true,
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  List<String> _list = [
    "text",
    "text",
    "text",
    "text",
    "text",
    "text",
    "text",
    "text",
    "text",
    "text",
    "text",
    "text",
    "text",
    "text",
    "text",
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Expanded(
              child: ListView.builder(
                itemCount: _list.length,
                itemBuilder: (context, index) => ItemLine(),
              ),
            )
          ],
        ),
      ),
    );
  }
}

class ItemLine extends StatelessWidget {
  const ItemLine({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Column(
      children: [
        Row(
          mainAxisAlignment: MainAxisAlignment.spaceBetween,
          children: [
            Expanded(
              child: Padding(
                padding: const EdgeInsets.all(8.0),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text("TEST"),
                    Row(
                      children: [
                        IconButton(
                          onPressed: () {},
                          icon: const Icon(
                            Icons.remove_circle_outline,
                          ),
                        ),
                        Text("1"),
                        IconButton(
                          onPressed: () {},
                          icon: const Icon(
                            Icons.add_circle,
                          ),
                        ),
                      ],
                    ),
                  ],
                ),
              ),
            ),
          ],
        ),
        const Divider()
      ],
    );
  }
}
```

</details>

**Comment by bernaferrari** [comment]
Do you have a reproducible code?

**Comment by Elblassy** [comment]
no, it work as i expect in IOS and when make material 3 is false it work fine so i expect that problem from material 3

**Comment by bernaferrari** [comment]
How can we believe in you if you don't have a sample?

**Comment by Elblassy** [comment]
i update issue with main.dart you can copy it and test it

**Comment by jonahwilliams** [comment]
I can repro this pretty easily. The application raster times are reasonable, and then after I hit a checkbox a dozen or so times they massively regress and stay that way.

**Comment by jonahwilliams** [comment]
It actually looks like the old ink sparkles might not be getting removed in this case, tentatively adding framework too

**Comment by jonahwilliams** [comment]
So it looks like the old ink splash's animation controller had a time limit, so we didn't need to worry about disposing them. The new ink sparkle runs forever, so if dispose isn't called we'll create a super expensive stack of drawPaints/saveLayers

**Comment by jonahwilliams** [comment]
Confirmed this can be "fixed" by adding `Future<void>.delayed(const Duration(seconds: 1)).whenComplete(dispose)` to the constructor of the ink sparkle to ensure it gets disposed

**Comment by jonahwilliams** [comment]
> So it looks like the old ink splash's animation controller had a time limit, so we didn't need to worry about disposing them. The new ink sparkle runs forever, so if dispose isn't called we'll create a super expensive stack of drawPaints/saveLayers

Actually I take that back, the new one has a time limit too...

**Comment by jonahwilliams** [comment]
Nah wasn't a timer issue, we were missing a status listener in the animation controller to handle disposing. Fix pending

**Comment by jongkb** [comment]
> Nah wasn't a timer issue, we were missing a status listener in the animation controller to handle disposing. Fix pending

Good to know. Fix would also address #104229.
